### PR TITLE
Update s3 gateway docs with recent changes

### LINKED ISSUE #4064: Update s3gateway docs
The s3gateway docs need to be updated to reflect the new features landed in https://github.com/pachyderm/pachyderm/pull/3906. Most of the API docs are updated, but [the guide](http://docs.pachyderm.io/en/latest/enterprise/s3gateway.html) needs some touchups.

### PR DESCRIPTION
**Pull request by ysimonson**
Closes #4064 

I also cut out the browser and curl example of bucket listing because it no longer works with auth mode, as well as some other minor touch-ups. The parts about authentication probably need the most vetting, as AWS' idiosyncratic authentication methodologies are not the most straight-forward to explain.

### CODE REVIEW

File: doc/reference/s3gateway_api.md [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
 
 Generally, you would not call these endpoints directly, but rather use a
 tool or library designed to work with S3-like APIs. Because of that, some
 working knowledge of S3 and HTTP is assumed.
 
+### Authentication
+
+If authentication is not enabled on the Pachyderm cluster, s3 gateway endpoints can be hit without passing auth credentials.
+
+If authentication is enabled, credentials must be passed using AWS' [signature v2](https://docs.aws.amazon.com/AmazonS3/latest/dev/RESTAuthentication.html) or [signature v4](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html) methods. For both methods, the AWS access key and secret key should be set to the same value; they are both the Pachyderm auth token used to issue the relevant PFS calls. Note that one or both signature methods are built into all s3 tools and libraries already, so you don't need to hand-roll these methods.
```
Comment (svekars) [inline doc/reference/s3gateway_api.md]:
```suggestion
If authentication is enabled, credentials must be passed using AWS' [signature v2](https://docs.aws.amazon.com/AmazonS3/latest/dev/RESTAuthentication.html) or [signature v4](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html) methods. For both methods, set the AWS access key and secret key to the same value. Both values are the Pachyderm auth token that is used to issue the relevant PFS calls. One or both signature methods are built into most s3 tools and libraries already, so you do not need to configure these methods manually.
```

File: doc/reference/s3gateway_api.md [hunk 2]
```diff
@@ -120,7 +126,7 @@ Aborts an in-progress multipart upload.
 
 Route: `POST /<branch>.<repo>?uploadId=<uploadId>`
 
-Completes a multipart upload. Note that if ETags are included in the request payload, they must be of the same format as returned by s3gateway when the multipart chunks are included. If they are md5 hashes (or any other hash algorithm), they will be ignored.
+Completes a multipart upload. Note that if ETags are included in the request payload, they must be of the same format as returned by s3 gateway when the multipart chunks are included. If they are md5 hashes (or any other hash algorithm), they will be ignored.
```
Comment (svekars) [inline doc/reference/s3gateway_api.md]:
```suggestion
Completes a multipart upload. If ETags are included in the request payload, they must be of the same format as returned by the s3 gateway when the multipart chunks are included. If they are `md5` hashes or any other hash algorithm, they are ignored.
```

File: doc/enterprise/s3gateway.md [hunk 3]
```diff
@@ -74,13 +61,13 @@ correspond to your S3 client.
...
 
-If you are using AWS CLI or S3cmd, skip this section.
+If you are not using the MinIO client, skip this section.
 
 To install and configure MinIO, complete the following steps:
 
 1. Install the MinIO client on your platform as
 described on the [MinIO download page](https://min.io/download#/macos).
-1. Verify that MinIO components are successfully installed by running
+2. Verify that MinIO components are successfully installed by running
```
[the remark this answers is not in the document]
Comment (ysimonson) [reply doc/enterprise/s3gateway.md]: This way the markdown source is more readable, which is how I'll often go through the documents. Happy to change it back though, if we want to use lazy lists as a convention.

[the remark this answers is not in the document]
Comment (svekars) [reply doc/enterprise/s3gateway.md]: Not a blocker, but would appreciate if you change it back =)

File: doc/enterprise/s3gateway.md [hunk 4]
```diff
@@ -3,68 +3,55 @@
...
 
-   ![S3 buckets](../_images/s_list_of_buckets.png)
-
-1. Alternatively, you can use `curl`:
-
-   ```
-   $ curl http://localhost:30600
-   <ListAllMyBucketsResult><Owner><ID>00000000000000000000000000000000</ID><DisplayName>pachyderm</DisplayName></Owner><Buckets><Bucket><Name>master. train</Name><CreationDate>2019-07-12T22:09:50.274391271Z</CreationDate></Bucket><Bucket><Name>master.pre_process</Name><CreationDate>2019-07-12T21:   58:50.930608352z</CreationDate></Bucket><Bucket><Name>master.split</Name><CreationDate>2019-07-12T21:58:09.074523275Z</CreationDate></                Bucket><Bucket><Name>stats.split</Name><CreationDate>2019-07-12T21:58:09.074523275Z</CreationDate></Bucket><Bucket><Name>master.raw_data</Name><CreationDate>2019-07-12T21:36:27.975670319Z</CreationDate></Bucket></Buckets></ListAllMyBucketsResult>
-   ```
+If you do not have direct access to the kubernetes cluster, you can use port
```
Comment (svekars) [inline doc/enterprise/s3gateway.md]:
```suggestion
If you do not have direct access to the Kubernetes cluster, you can use port
```

File: doc/enterprise/s3gateway.md [hunk 5]
```diff
@@ -3,68 +3,55 @@
...
-   For this to work, enable
-   port forwarding by running the `pachctl port-forward` command.
-
-   However, Pachyderm does not recommend to heavily rely on port forwarding.
-   Because Kubernetes' port forwarder incurs overhead, it might
-   not recover well from broken connections. Therefore, using
-   Pachyderm contexts or connecting to the S3 gateway directly
-   through the cluster IP address are more reliable and preferred
-   options.
+However, note that kubernetes' port forwarder incurs substantial overhead and
```
Comment (svekars) [inline doc/enterprise/s3gateway.md]:
```suggestion
However, the Kubernetes port forwarder incurs substantial overhead and
```

**Comment by ysimonson** [comment]
We need to standardize on naming. I've tentatively changed the API docs to go with convention in the user docs (`S3 gateway`), rather than the original `s3gateway`. Is this what we want?

**Pull request merged**